custom_components/tech/select.py

```diff
@@ -0,0 +1,283 @@
+
+from . import assets
+from .const import (
+    CONTROLLER,
+    DOMAIN,
+    FAN_MODE_IDO_ID,
+    FAN_MODE_OPTIONS,
+    FAN_MODE_OPTIONS_REVERSE,
+    GEAR_CONTROL_IDO_ID,
+    GEAR_OPTIONS,
+    GEAR_OPTIONS_REVERSE,
+    INCLUDE_HUB_IN_NAME,
+    MANUFACTURER,
+    RECUPERATION_EXHAUST_FLOW,
+    RECUPERATION_SUPPLY_FLOW,
+    RECUPERATION_SUPPLY_FLOW_ALT,
+    TYPE_TEMPERATURE_CH,
+    UDID,
+    VER,
+)
+from .coordinator import TechCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up entry."""
+    _LOGGER.debug(
+        "Setting up select entry, controller udid: %s",
+        config_entry.data[CONTROLLER][UDID],
+    )
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+    controller_udid = config_entry.data[CONTROLLER][UDID]
+
+    tiles = await coordinator.api.get_module_tiles(controller_udid)
+
+    entities = []
+
+    # Check if we have recuperation system (detected by flow sensors)
+    has_recuperation_flow = False
+    for t in tiles:
+        tile = tiles[t]
+        if tile.get("type") == TYPE_TEMPERATURE_CH:
+            widget1_txt_id = tile.get("params", {}).get("widget1", {}).get("txtId", 0)
+            widget2_txt_id = tile.get("params", {}).get("widget2", {}).get("txtId", 0)
+            for flow_sensor in [RECUPERATION_EXHAUST_FLOW, RECUPERATION_SUPPLY_FLOW, RECUPERATION_SUPPLY_FLOW_ALT]:
+                if flow_sensor["txt_id"] in [widget1_txt_id, widget2_txt_id]:
+                    has_recuperation_flow = True
+                    break
+        if has_recuperation_flow:
+            break
+
+    # Create select controls if we have recuperation
+    if has_recuperation_flow:
+        _LOGGER.debug("Creating recuperation select controls")
+        entities.append(RecuperationGearSelect(coordinator, config_entry))  # Direct gear control
+        entities.append(RecuperationModeSelect(coordinator, config_entry))  # Timed fan mode
+    else:
+        _LOGGER.debug("No recuperation flow detected, skipping select controls")
+
+    async_add_entities(entities, True)
+
+
+class RecuperationModeSelect(CoordinatorEntity, SelectEntity):
+    """Representation of recuperation fan mode select control (timed mode changes)."""
+
+    _attr_has_entity_name = True
+    _attr_icon = "mdi:fan-speed"
+
+    def __init__(
+        self,
+        coordinator: TechCoordinator,
+        config_entry: ConfigEntry,
+    ) -> None:
+        """Initialize the fan mode select."""
+        super().__init__(coordinator)
+        self._coordinator = coordinator
+        self._config_entry = config_entry
+        self._udid = config_entry.data[CONTROLLER][UDID]
+        self._attr_unique_id = f"{self._udid}_recuperation_fan_mode"
+
+        # Define options with Polish names
+        self._attr_options = [
+            "Zatrzymaj wentylator",
```

Comment:
We should support all languages from tech API as minimum:
```
Options: 'en', 'fr', 'it', 'es', 'nl', 'pl', 'de', 'cs', 'sk', 'hu', 'ro', 'lt', 'et', 'ru', 'si', 'hr'
```

As we got a users from those countries also

Reply:
Yes, this is all in the language file returned by the API:
https://emodul.eu/api/v1/i18n/en
https://emodul.eu/api/v1/i18n/pl
<img width="334" height="25" alt="image" src="https://github.com/user-attachments/assets/58d2751c-c575-4304-a123-3573b81cc0b6" />
<img width="212" height="36" alt="image" src="https://github.com/user-attachments/assets/b76c064d-2b1f-476a-baa1-32c919eae938" />
so you need to use the key from there